pkg/volume/glusterfs/glusterfs.go

```diff
@@ -54,8 +56,18 @@ func (plugin *glusterfsPlugin) Name() string {
 }
 
 func (plugin *glusterfsPlugin) CanSupport(spec *volume.Spec) bool {
-	return (spec.PersistentVolume != nil && spec.PersistentVolume.Spec.Glusterfs != nil) ||
-		(spec.Volume != nil && spec.Volume.Glusterfs != nil)
+	if (spec.PersistentVolume != nil && spec.PersistentVolume.Spec.Glusterfs == nil) ||
+		(spec.Volume != nil && spec.Volume.Glusterfs == nil) {
+		return false
+	}
+	// see if glusterfs mount helper is there
+	_, err := plugin.execCommand("ls", []string{"/sbin/mount.glusterfs"})
```

Comment:
Add a comment here that this needs a ls because the plugin container may be on a filesystem that is not visible to the volume plugin process.